app/pages/settings/email.jsx

```diff
@@ -5,11 +5,133 @@ import talkClient from 'panoptes-client/lib/talk-client';
+  return (
+    (talkPreferences.length > 0) ?
+      <tbody>
+        {talkPreferences.map((pref, i) => {
+          if (pref.category !== 'system' && pref.category !== 'moderation_reports') {
+            return (
+              <tr key={pref.id}>
+                <Translate component="td" content={`emailSettings.talk.options.${pref.category}`} />
+                <TalkPreferenceOption preference={pref} index={i} digest="immediate" onChange={onChange} />
+                <TalkPreferenceOption preference={pref} index={i} digest="daily" onChange={onChange} />
+                <TalkPreferenceOption preference={pref} index={i} digest="weekly" onChange={onChange} />
+                <TalkPreferenceOption preference={pref} index={i} digest="never" onChange={onChange} />
+              </tr>
+            );
+          } else {
+            return null;
+          }
+        })}
+      </tbody> :
+      <tbody />
+  );
+}
+
+TalkPreferences.propTypes = {
+  onChange: React.PropTypes.func,
+  talkPreferences: React.PropTypes.arrayOf(React.PropTypes.object).isRequired
+};
+
+function ProjectPreferences({ projects, projectPreferences, onChange }) {
+  return (
+    <tbody>
+      {projectPreferences.map((projectPreference, i) => {
+        if (projects[i]) {
+          return (
+            <tr key={projectPreference.id}>
+              <td>
+                <input
+                  type="checkbox"
+                  name="email_communication"
+                  checked={projectPreference.email_communication}
+                  onChange={(e) => onChange(i, e)}
+                />
+              </td>
+              <td>
+                {projects[i].display_name}
+              </td>
+            </tr>
+          );
+        } else {
+          return null;
+        }
+      })}
+    </tbody>
+  );
+}
+
+ProjectPreferences.propTypes = {
+  projects: React.PropTypes.arrayOf(React.PropTypes.object).isRequired,
+  projectPreferences: React.PropTypes.arrayOf(React.PropTypes.object).isRequired,
+  onChange: React.PropTypes.func
+};
+
+function Pagination({ meta, page, onChange }) {
+  if (meta) {
+    return (
+      <nav className="pagination">{'Page'}
+        <select
+          value={page}
+          disabled={meta.page_count < 2}
+          onChange={onChange}
+        >
+          {Array.from(Array(meta.page_count), (p, i) => {
+            return <option key={i} value={i + 1}>{i + 1}</option>;
+          })}
+        </select> {' of '} {meta.page_count || '?'}
+      </nav>
+    );
+  } else {
+    return null;
+  }
+}
+
+Pagination.propTypes = {
+  meta: React.PropTypes.shape({
+    page_count: React.PropTypes.number
+  }),
+  page: React.PropTypes.string,
+  onChange: React.PropTypes.func
+};
+
+Pagination.defaultProps = {
+  meta: {
+    page_count: 1
+  },
+  page: '1'
+}
+
 class EmailSettingsPage extends React.Component {
   constructor(props) {
     super(props);
     this.state = {
-      page: 1,
+      page: '1',
```

Comment:
Why a string? It was an integer in [the original](https://github.com/zooniverse/Panoptes-Front-End/blob/master/app/pages/settings/email.cjsx#L16)

Reply:
Try changing it to `React.PropTypes.number` and watch the invalid prop warnings pile up in the dev console.

Reply:
Or change this back to a number and watch the warnings appear when you run the tests.

Reply:
Mm that's annoying, thanks.

Reply:
Do you want me to fix the email settings component so that page number is always passed around as a number?

Reply:
Fixed in 2fa1281